### ISSUE TITLE
Server error when browsing admin trough the API.

### ISSUE THREAD

**Issue by vstoykov** [wagtail/wagtail#3863]
### Issue Summary

Server error when browsing admin trough the API and there was stale content type(s) (because app is removed or branch is changed during development). It is similar to the error that was fixed by #3569 but this is for the API.

### Steps to Reproduce

1. Create some app with page model extending from `Page`
2. Add a page of the time from the new app.
3. Commit the changes in new branch and then go back to master, or just remove the app from `INSTALLED_APPS`
4. From Admin start navigating inside root page by clicking the right arrow from left menu (pages).

It should display list of pages under the root page bit instead will show `Server error`.

### Technical details

* Python version: 3.6
* Django version: 1.11.5
* Wagtail version: 1.12.1
* Browser version: Chrome 61

Traceback:
```
Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/django/core/handlers/exception.py", line 41, in inner
    response = get_response(request)
  File "/usr/lib/python3.6/site-packages/django/core/handlers/base.py", line 187, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/usr/lib/python3.6/site-packages/django/core/handlers/base.py", line 185, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/lib/python3.6/site-packages/django/views/decorators/cache.py", line 43, in _cache_controlled
    response = viewfunc(request, *args, **kw)
  File "/usr/lib/python3.6/site-packages/wagtail/wagtailadmin/decorators.py", line 31, in decorated_view
    return view_func(request, *args, **kwargs)
  File "/usr/lib/python3.6/site-packages/wagtail/api/v2/router.py", line 65, in wrapped
    return func(request, *args, **kwargs)
  File "/usr/lib/python3.6/site-packages/django/views/decorators/csrf.py", line 58, in wrapped_view
    return view_func(*args, **kwargs)
  File "/usr/lib/python3.6/site-packages/rest_framework/viewsets.py", line 90, in view
    return self.dispatch(request, *args, **kwargs)
  File "/usr/lib/python3.6/site-packages/rest_framework/views.py", line 489, in dispatch
    response = self.handle_exception(exc)
  File "/usr/lib/python3.6/site-packages/wagtail/api/v2/endpoints.py", line 95, in handle_exception
    return super(BaseAPIEndpoint, self).handle_exception(exc)
  File "/usr/lib/python3.6/site-packages/rest_framework/views.py", line 449, in handle_exception
    self.raise_uncaught_exception(exc)
  File "/usr/lib/python3.6/site-packages/rest_framework/views.py", line 486, in dispatch
    response = handler(request, *args, **kwargs)
  File "/usr/lib/python3.6/site-packages/wagtail/wagtailadmin/api/endpoints.py", line 93, in listing_view
    response = super(PagesAdminAPIEndpoint, self).listing_view(request)
  File "/usr/lib/python3.6/site-packages/wagtail/api/v2/endpoints.py", line 81, in listing_view
    return self.get_paginated_response(serializer.data)
  File "/usr/lib/python3.6/site-packages/rest_framework/serializers.py", line 739, in data
    ret = super(ListSerializer, self).data
  File "/usr/lib/python3.6/site-packages/rest_framework/serializers.py", line 263, in data
    self._data = self.to_representation(self.instance)
  File "/usr/lib/python3.6/site-packages/rest_framework/serializers.py", line 657, in to_representation
    self.child.to_representation(item) for item in iterable
  File "/usr/lib/python3.6/site-packages/rest_framework/serializers.py", line 657, in <listcomp>
    self.child.to_representation(item) for item in iterable
  File "/usr/lib/python3.6/site-packages/wagtail/api/v2/serializers.py", line 284, in to_representation
    meta[field.field_name] = field.to_representation(attribute)
  File "/usr/lib/python3.6/site-packages/wagtail/api/v2/serializers.py", line 91, in to_representation
    name = page.specific_class._meta.app_label + '.' + page.specific_class.__name__
AttributeError: 'NoneType' object has no attribute '_meta'
```

**Comment by gregtlv** [comment]
I have same problem

**Comment by Pomax** [comment]
Now that wagtail 2.0 is officially out, upgrading immediately exposes this behaviour for established wagtail installations, too.

Navigating by clicking `pages` and then clicking on `>` to expand the tree causes a crash with tracelog in the console, and a "server error" client side.

**Comment by gasman** [comment]
@Pomax Does your stack trace match the one that @vstoykov gives above? Upgrading certainly shouldn't cause any missing model definitions, so I'd be cautious about assuming that this is the same issue.

**Comment by Pomax** [comment]
it did, however it turns out that this was because of a db mismatch between a dev branch with migrations from 1, used in the upgrade branch for 2. After moving the dev branch to 2 as well, the error goes away so this was a false alarm.

**Comment by jawinn** [comment]
I received the same error after running a migration that deleted a custom Page model, when there was an existing page using that model under Home. Pages list shows "Server Error" after expanding Home.

**Comment by axper** [comment]
@Pomax Can you please tell more details on how you solved this issue? I'm getting the same kind of stack trace here.

**Comment by Pomax** [comment]
it really was a matter of having both a dev and prod instance that both needed to run v2 rather than having one set to v2, and the other to v1, with migrations that belonged to v1 being used with the v2 instance. The solution was to first upgrade the v1 instance to v2, making sure to fix migrations so they pointed to the new app names instead of the old app names.

**Comment by axper** [comment]
Upgrading to latest Wagtail 2.2 version fixed this for me.

**Comment by gasman** [comment]
Fixed in #4648
